flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/Executor.java

```diff
@@ -32,13 +32,13 @@
     void start() throws SqlExecutionException;
 
     /**
-     * Open a new session by using the given {@link SessionContext}.
+     * Open a new session by using the given session id.
      *
-     * @param session context to create new session.
-     * @return session identifier to track the session.
+     * @param sessionId session identifier.
+     * @return used session identifier to track the session.
      * @throws SqlExecutionException if any error happen
      */
-    String openSession(SessionContext session) throws SqlExecutionException;
+    String openSession(String sessionId) throws SqlExecutionException;
```

Comment:
Currently, we call it in `SqlClient` like this `executor.openSession(options.getSessionId())`, but `options.getSessionId()` might be `null`. I suggest either add `@Nullable` annotation to the parameter, or construct a not-null `sessionId` in `SqlClient` and pass it to `openSession(..)`.